lighthouse-core/gather/gatherers/fonts.js

```diff
@@ -0,0 +1,158 @@
+            display: rule.style.fontDisplay || 'auto',
+            family: rule.style.fontFamily.replace(/"|'/g, ''),
+            stretch: rule.style.fontStretch || 'normal',
+            style: rule.style.fontStyle || 'normal',
+            weight: rule.style.fontWeight || 'normal',
+            variant: rule.style.fontVariant || 'normal',
+            unicodeRange: rule.style.unicodeRange || 'U+0-10FFFF',
+            featureSettings: rule.style.featureSettings || 'normal',
+            src: [],
+          };
+
+          if (rule.style.src) {
+            const matches = rule.style.src.match(new RegExp(fontUrlRegex, 'g'));
+            if (matches) {
+              fontsObject.src = matches.map(match => {
+                const res = new RegExp(fontUrlRegex).exec(match);
+                return new URL(res[1], location.href).href;
+              });
+            }
+          }
+
+          fontFaceRules.push(fontsObject);
+        }
+      }
+    }
+
+    return fontFaceRules;
+  }
+
+  /**
+   * Provided a <link rel=stylesheet> element, it attempts to reload the asset with CORS headers.
+   * Without CORS headers, a cross-origin stylesheet will have node.styleSheet.cssRules === null.
+   * @param {Element} oldNode
+   * @return {<!Promise>}
+   */
+  function loadStylesheetWithCORS(oldNode) {
+    const newNode = oldNode.cloneNode(true);
+
+    return new Promise(resolve => {
+      newNode.addEventListener('load', function onload() {
+        newNode.removeEventListener('load', onload);
+        resolve(getFontFaceFromStylesheets());
+      });
+      newNode.crossOrigin = 'anonymous';
+      oldNode.parentNode.insertBefore(newNode, oldNode);
+      oldNode.remove();
+    });
+  }
+
+  const promises = [];
+  // Get all loaded stylesheets
+  for (const stylesheet of document.styleSheets) {
+    try {
+      // Cross-origin stylesheets don't expose cssRules by default. We reload them w/ CORS headers.
+      if (stylesheet.cssRules === null && stylesheet.href && stylesheet.ownerNode &&
+        !stylesheet.ownerNode.crossOrigin) {
+        promises.push(loadStylesheetWithCORS(stylesheet.ownerNode));
+      } else {
+        promises.push(Promise.resolve(getSheetsFontFaces(stylesheet)));
+      }
+    } catch (err) {
+      promises.push(loadStylesheetWithCORS(stylesheet.ownerNode));
+    }
+  }
+  // Flatten results
+  return Promise.all(promises).then(fontFaces => [].concat(...fontFaces));
+}
+/* eslint-env node */
+
+class Fonts extends Gatherer {
+  _findSameFontFamily(fontFace, fontFacesList) {
+    return fontFacesList.find(fontItem => {
+      return !fontFaceDescriptors.find(descriptor => {
+        return fontFace[descriptor] !== fontItem[descriptor];
+      });
+    });
+  }
+
+  afterPass({driver}) {
+    return Promise.all(
+      [
+        driver.evaluateAsync(`(()=>{`
+          + `const args = ${JSON.stringify(fontFaceDescriptors)};`
+          + `return (${getAllLoadedFonts.toString()})(args);})()`),
```

Comment:
could stringify outside the eval just to make the call as similar as possible to the other :)

```js
const args = JSON.stringify(...)
return Promise.all([
  driver.evaluateAsync(`(${getAllLoadedFonts.toString()})(${args})`),
  driver.evaluateAsync(`(${getFontFaceFromStylesheets.toString()})()`),